lib/solarwinds_apm/otel_native_config.rb

```diff
@@ -0,0 +1,97 @@
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+require 'solarwinds_apm/constants'
+require 'solarwinds_apm/api'
+require 'solarwinds_apm/support'
+require 'solarwinds_apm/opentelemetry'
+require 'solarwinds_apm/sampling'
+
+module SolarWindsAPM
+  # OTelNativeConfig module
+  module OTelNativeConfig
+    @@config           = {}
+    @@config_map       = {}
+    @@agent_enabled    = true
+
+    def self.initialize
+      return unless defined?(::OpenTelemetry::SDK::Configurator)
+
+      ENV['OTEL_TRACES_EXPORTER'] = ENV['OTEL_TRACES_EXPORTER'].to_s.split(',').tap { |e| e << 'otlp' unless e.include?('otlp') }.join(',')
+      ENV['OTEL_RESOURCE_ATTRIBUTES'] = "sw.apm.version=#{SolarWindsAPM::Version::STRING},sw.data.module=apm,service.name=#{ENV.fetch('OTEL_SERVICE_NAME', nil)}," + ENV['OTEL_RESOURCE_ATTRIBUTES'].to_s
+
+      # add response propagator to rack instrumentation
+      resolve_response_propagator
+
+      # dbo: traceparent injection as sql comments
+      require_relative 'patch/tag_sql_patch' if SolarWindsAPM::Config[:tag_sql]
+
+      # sdk config will initialize trace and metrics exporter
+      # any setup on endpoint for metrics and trace exporter should happen here
+      # also set to exporter to console for testing purpose
+      ::OpenTelemetry::SDK.configure { |c| c.use_all(@@config_map) }
+
+      # append our propagators
+      ::OpenTelemetry.propagation.instance_variable_get(:@propagators).append(SolarWindsAPM::OpenTelemetry::SolarWindsPropagator::TextMapPropagator.new)
+
+      # add sw metrics processors (only record respone_time)
+      txn_manager = TxnNameManager.new
+      otlp_processor = SolarWindsAPM::OpenTelemetry::OTLPProcessor.new(txn_manager)
+      @@config[:metrics_processor] = otlp_processor
+      ::OpenTelemetry.tracer_provider.add_span_processor(otlp_processor)
+
+      service_key_name = ENV['SW_APM_SERVICE_KEY'].to_s.split(':')
+
+      # no need to send init msg for otlp proto
+      # need to consider endpoint for get setting and endpoint for otlp exporters
+      # current implementation only use OTEL env for endpoint
+      #             -> need to come up with logic to drive from SW_APM_COLLECTOR
+      sampler_config = {
+        collector: "https://#{ENV.fetch('SW_APM_COLLECTOR', 'apm.collector.cloud.solarwinds.com')}:443",
+        service: service_key_name[1],
+        headers: "Bearer #{service_key_name[0]}",
+        tracing_mode: SolarWindsAPM::Config[:tracing_mode],
+        trigger_trace_enabled: SolarWindsAPM::Config[:trigger_tracing_mode],
+        transaction_settings: SolarWindsAPM::Config[:transaction_settings]
+      }
+
+      # configure sampler afterwards
+      http_sampler = HttpSampler.new(sampler_config)
```

Comment:
When would the new `JsonSampler` be used instead?

Reply:
I haven't start implement the lambda version. I will update it.